zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -657,25 +656,27 @@ where
     // Store the transactions only after creating all of them. This avoids undesired
     // retransmissions in case a transaction is stored and the creation of a subsequent
     // transaction fails.
+    let mut transactions = Vec::with_capacity(proposal.steps().len());
     let mut txids = Vec::with_capacity(proposal.steps().len());
     #[allow(unused_variables)]
-    for (_, (build_result, outputs, fee_amount, utxos_spent)) in step_results {
+    for (_, (build_result, outputs, fee_amount, utxos_spent)) in step_results.iter() {
         let tx = build_result.transaction();
-        let sent_tx = SentTransaction {
+        transactions.push(SentTransaction {
```

Comment:
This could alternatively use `SentTransaction::new`, and its fields could be private.